added tunnel icon

jcritchley opened the pull request:
Adding a tunnel icon. Closes issue [#475](https://github.com/mapbox/maki/issues/475).

<img width="488" alt="image" src="https://user-images.githubusercontent.com/2847668/171881452-18590723-b822-4b70-a0e2-a10d1ca9971b.png">

tristen reviewed (changes requested):
@jcritchley sweet! I think this communicates a tunnel really well. A couple of thoughts:


1. Most Maki icons ~1px spacing between the edges. See `square` as example:

<img width="1440" alt="Screen Shot 2022-06-03 at 3 45 33 PM" src="https://user-images.githubusercontent.com/61150/171939335-7b469476-908e-4526-afae-db41d20c8773.png">
 
This makes the icon fit a tad more comfortably against a round or square background. Thoughts on shaving off a couple a bit of the shape for space?

---

0.5px rounding is inconsistent with 1px we use (compare with `square`). Could you bring the path nodes up to the nearest whole value?

<img width="1440" alt="Screen Shot 2022-06-03 at 3 45 10 PM" src="https://user-images.githubusercontent.com/61150/171939831-1ee3cb33-0759-44b5-be35-f2625598a153.png">

---

3.  I'm wondering if the inside rounding is necessary here. Especially after moving to 1px it might look a little too rounded. Thoughts on dropping the inside rounding entirely?

<img width="565" alt="Screen Shot 2022-06-03 at 3 52 03 PM" src="https://user-images.githubusercontent.com/61150/171940477-b8625939-2be0-4724-af2f-9e53cf5440c7.png">

jcritchley commented:
@tristen thanks for the feedback! I've updated incorporating changes reducing the spacing and fixing the bottom outer corner rounding to 1px. 

<img width="201" alt="image" src="https://user-images.githubusercontent.com/2847668/171954543-59d31e6b-9982-49dc-8822-95941cb0687c.png">

Original designs did not include the internal rounding and I added it at the last minute, but agree it's not necessary and there's better alignment at the bottom when using a stroke when squared:

![image](https://user-images.githubusercontent.com/2847668/171953405-bf39be24-8e01-4cd9-b751-abd357a2d428.png)

I think the internal area works better without rounding as well but curious on other thoughts, here it is with the original 0.5px rounding:

<img width="231" alt="image" src="https://user-images.githubusercontent.com/2847668/171955129-dfdef5ea-e925-4403-b7a9-6d897219a953.png">

I was trying to keep a similar height to the original but extending the internal arc can make it feel stretched or left with excessive fill at the top:

<img width="201" alt="image" src="https://user-images.githubusercontent.com/2847668/171954109-3900af1a-daec-417d-bd4f-53e4b5f5974d.png">

tristen reviewed (approved):
The changes look great to me ⚡ 

> I think the internal area works better without rounding as well but curious on other thoughts, here it is with the original 0.5px rounding:

I'll let @willymaps weigh in on this too but I agree the internal area looks pretty clean without rounding.

willymaps reviewed (approved):
Looks great @jcritchley!
